### PR TITLE
Feature/accounting command

### PR DESCRIPTION
**Pull request by chihkaiyu**
Changes:  
- Implement cost command
- Remove db instance from bot API
- Add `kingpin` to dependency

### CODE REVIEW

File: command/message.go [hunk 1]
```diff
@@ -2,23 +2,77 @@ package command
...
+	return c, nil
 }
 
 func (c *messageCMD) Command(ctx context.Context, event *linebot.Event) ([]linebot.Message, error) {
-	m := linebot.NewTextMessage("test message")
-	return []linebot.Message{m}, nil
+	switch message := event.Message.(type) {
+	case *linebot.TextMessage:
+		splitedCommand := strings.Split(message.Text, " ")
+		if splitedCommand[0] == "cost" || splitedCommand[0] == "Cost" {
```
Comment (jiazhen-lin) [inline command/message.go]: We should make string lower case and check 
ex: "COST"

File: command/message.go [hunk 2]
```diff
@@ -2,23 +2,77 @@ package command
...
+	kingpin "gopkg.in/alecthomas/kingpin.v2"
 )
 
 type messageCMD struct {
+	db *sqlx.DB
+}
+
+type costParam struct {
+	price          float64
+	category       string
```
Comment (jiazhen-lin) [inline command/message.go]: Use enum instead of string

File: command/message.go [hunk 3]
```diff
@@ -2,23 +2,77 @@ package command
...
+			// parse cost command
+			param, err := parseCostCommand(splitedCommand[1:])
+			if err != nil {
+				m := linebot.NewTextMessage(err.Error())
+				return []linebot.Message{m}, nil
+			}
+			userID := event.Source.UserID
+
+			// insert new record to db
+			insert := "INSERT INTO Accounting (userID, category, accountingTime, price, detail) VALUES (?, ?, ?, ?)"
```
Comment (jiazhen-lin) [inline command/message.go]: You have 5 parameters in statement but only 4 ?

File: command/message.go [hunk 4]
```diff
@@ -2,23 +2,77 @@ package command
...
+			_, err = c.db.Exec(insert, userID, param.category, param.accountingTime, param.price, param.detail)
+			if err != nil {
+				m := linebot.NewTextMessage(err.Error())
+				return []linebot.Message{m}, nil
+			}
+			m := linebot.NewTextMessage("Accounting recorded!")
+			return []linebot.Message{m}, nil
+		}
+		m := linebot.NewTextMessage("Unknown command!\n" + costUsage)
```
Comment (jiazhen-lin) [inline command/message.go]: Why return costUsage to user?

Comment (chihkaiyu) [reply command/message.go]: Tell them how to use this command.

File: command/message.go (hunk 1 above)
Comment (chihkaiyu) [reply command/message.go]: Done.

File: command/message.go (hunk 2 above)
Comment (chihkaiyu) [reply command/message.go]: Done.

File: command/message.go (hunk 3 above)
Comment (chihkaiyu) [reply command/message.go]: Done T.T

**Pull request closed without merging**